addons/survey/models/survey_survey.py

```diff
@@ -552,11 +681,36 @@ def action_survey_preview_certification_template(self):
             'url': '/survey/%s/get_certification_preview' % (self.id)
         }
 
+    def action_start_session(self):
+        """ Sets the necessary fields for the session to take place and starts it. """
+        self.ensure_one()
+        self.write({
+            'questions_layout': 'page_per_question',
+            'session_current_question_id': None,
+            'session_state': 'ready'
+        })
+        return self.action_open_session_manager()
+
+    def action_open_session_manager(self):
+        self.ensure_one()
+
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Open Session Manager",
+            'target': 'self',
+            'url': '/survey/session_manage/%s' % self.access_token
+        }
+
+    def action_end_session(self):
+        self.write({'session_state': 'not_started'})
```

Comment:
I guess this can have also self.ensure_one. But this could be useful if we can use this action in a tree view (so with multiple selected records)

Reply:
I don't really see the use case where you would want to end several survey sessions at the same time.
So let's go with `ensure_one` :D

Reply:
Why doing an ensure one if not really necessary ? Odoo is batch !